peachjam_search/models.py

```diff
@@ -77,3 +84,56 @@ class Meta:
     def save(self, *args, **kwargs):
         self.score = exp(-0.1733 * (self.position - 1))
         super().save(*args, **kwargs)
+
+
+class SavedSearch(models.Model):
+    q = models.CharField(max_length=4098)
+    filters = models.CharField(max_length=4098)
+    note = models.TextField(null=True, blank=True)
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    created_at = models.DateField(auto_now_add=True)
+    last_alert = models.DateTimeField(null=True, blank=True)
+
+    def __str__(self):
+        return self.q
+
+    def update_and_alert(self):
+        hits = self.find_new_hits()
+        if hits:
+            self.send_alert(hits)
+            self.last_alert = now()
+            self.save()
+
+    def find_new_hits(self):
+        from peachjam_search.views import DocumentSearchViewSet
+
+        factory = APIRequestFactory()
+        request = factory.get("/search/api/documents/")
+        request.user = self.user
+        params = f"q={self.q}&{self.filters}&created_at__gte={self.last_alert}"
+        request.GET = QueryDict(params)
+        view = DocumentSearchViewSet.as_view({"get": "list"})
+        response = view(request)
+        hits = response.data["results"]
+        return hits
+
+    def send_alert(self, hits):
+        # send an email or other alert to the user
+        hits = hits[:10]
+        context = {
+            "hits": hits,
+            "saved_search": self,
+            "site": Site.objects.get_current(),
+        }
+        html = render_to_string("peachjam_search/emails/search_alert.html", context)
+        plain_txt = render_to_string("peachjam_search/emails/search_alert.txt", context)
+
+        subject = settings.EMAIL_SUBJECT_PREFIX + _("New hits for search ") + self.q
```

Comment:
```suggestion
        subject = settings.EMAIL_SUBJECT_PREFIX + _("New matches for search: ") + self.q
```

Reply:
The translation is an interesting consideration here, because we don't have a request to know what language to use. Ideally, we should store (and use) the user's preferred language, but we don't have that currently.

I guess this will use the default site language for now, which is fine (we only have English sites any way).

Reply:
noted, subject updated